Fix scrolling

stonerl opened the pull request:
Implemented diagonal scrolling:
* Number pad: 1/3/7/9 scrolls diagonal. 
* 5 works like pressing HOME.
* Arrow keys: LEFT/UP, LEFT/DOWN, RIGHT/UP & RIGHT/DOWN scrolls diagonal

Scrolling distance is 1/8, with SHIFT 1/4 and with CTRL 1/16 of the screen resolution in x/y-direction and set to smooth.
fixes #3351 
fixes #3429

stonerl committed:
fix diagonal scrolling

stonerl committed:
fix scrolling

* move scrolling back to InteractiveBase::think()
* Windows don't focus() by default
* minimzed windows don't focus anymore

stonerl committed:
regain focus on restore

Regain focus when restoring for:

* Game Message Menu
* Seafaring Statistics Window

stonerl committed:
Regain Focus for Main Save Window

stonerl committed:
Merge branch 'master' into 3603-fix-scrolling

stonerl committed:
Revert "fix diagonal scrolling"

This reverts commit 6826a91e16368207c222a557de6da23cc2822a11.

stonerl committed:
Revert "fix scrolling"

This reverts commit 587db3b35df7d81593d54c6f711b0e899d87b46b.

stonerl committed:
Revert "Regain Focus for Main Save Window"

This reverts commit 5951ebe9703584c6ed76e9951a8283b1ba874c69.

stonerl committed:
Merge branch 'master' into 3603-fix-scrolling

stonerl committed:
Revert "regain focus on restore"

This reverts commit 2206f9ce41e7400327ca86d5ef232a8c9c21cbf3.

stonerl committed:
Add diagonal scrolling to keypad:

1/3/7/9 scroll diagonal, 5 returns to HQ.

stonerl committed:
change scrolling distance

Scrolling distance is now 1/4 auf the x/y-resolution.

Diagonal scrolling with arrow-key possible again.

stonerl committed:
pan_by() now needs transition attribute

stonerl committed:
move map scrolling to mapview.cc

stonerl committed:
numpad 1-9 only handles map scrolling

stonerl committed:
only scroll when numpad is off

Noordfrees reviewed (commented):
Code looking good, one comment inline. Will test this today…

Noordfrees commented:
Tested scrolling with the arrow keys and it works. (My keyboard doesn´t have a numpad so I can´t test that.) But the scrolling speed is much too high now IMHO. Please half the scrolling distance; and perhaps add Shift/Ctrl modifiers for faster/slower scrolling?

stonerl committed:
addressed code review

stonerl commented:
@Noordfrees I addressed your comments. Since I don't have a keypad myself we might need someone to test this. Although it should be working 😄

Noordfrees reviewed (dismissed):
Tested and working well.
Three more suggestions inline.

stonerl committed:
more code review fixes

stonerl commented:
Addressed your comments. @nifker do you have a keypad on your keyboard?

stonerl committed:
Merge branch 'master' into 3603-fix-scrolling

stonerl committed:
Merge branch 'master' into 3603-fix-scrolling

stonerl commented:
Thank you for your review.

ghost commented:
> Addressed your comments. @nifker do you have a keypad on your keyboard?

Yes I do and it worked - I just couldn't test the HOME button thingy.

stonerl commented:
> > Addressed your comments. @nifker do you have a keypad on your keyboard?
> 
> Yes I do and it worked - I just couldn't test the HOME button thingy.

Pressing 5 on the Keypad or the HOME button, should center your headquarter.

ghost commented:
Ah ok - yes thats working too.
